### PR TITLE
Update to the latest versions available

### PR DESCRIPTION
**Pull request by jborean93**
This updates the packages to the latest version available, also removes more packages at the end that aren't needed at runtime. The size of the package has been reduced from 96MB to 56MB.

This still does not match the version hosted at httpbin.org but there is a [bug](https://github.com/postmanlabs/httpbin/issues/617) in that implementation that would break our tests. Until that is resolved we shouldn't update any further.

### CODE REVIEW

**Comment by jborean93** [comment]
Still need further testing, looks like setuptools is requires for gunicorn.

**Comment by jborean93** [comment]
Requires https://github.com/ansible/ansible/pull/72013 for the tests to pass with the new test container.

**Review by mattclay — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -31,8 +32,11 @@ RUN set -x && \
     cp /root/ca/cacert.pem /usr/share/nginx/html/cacert.pem && \
     cp /root/ca/client.ansible.http.tests-cert.pem /usr/share/nginx/html/client.pem && \
     cp /root/ca/private/client.ansible.http.tests-key.pem /usr/share/nginx/html/client.key && \
-    pip install -c /root/constraints.txt gunicorn httpbin && \
-    apk del openssl-dev python2-dev libffi-dev gcc libstdc++ make musl-dev
+    chmod 644 /usr/share/nginx/html/* && \
+    pip install --no-cache-dir --no-compile -c /root/constraints.txt gunicorn httpbin && \
+    apk del openssl-dev py3-pip py3-wheel python3-dev libffi-dev gcc libstdc++ make musl-dev && \
+    rm -rf /root/.cache/pip && \
+    find /usr/lib/python3.8 -type f -regex ".*\.py[co]" -delete
```
Comment (sivel) [inline Dockerfile]: We probably also need to do:

```
find /usr/lib/python3.8 -type d -name "__pycache__" -delete
```

File: Dockerfile [hunk 2]
```diff
@@ -31,8 +32,11 @@ RUN set -x && \
     cp /root/ca/cacert.pem /usr/share/nginx/html/cacert.pem && \
     cp /root/ca/client.ansible.http.tests-cert.pem /usr/share/nginx/html/client.pem && \
     cp /root/ca/private/client.ansible.http.tests-key.pem /usr/share/nginx/html/client.key && \
-    pip install -c /root/constraints.txt gunicorn httpbin && \
-    apk del openssl-dev python2-dev libffi-dev gcc libstdc++ make musl-dev
+    chmod 644 /usr/share/nginx/html/* && \
+    pip install --no-cache-dir --no-compile -c /root/constraints.txt gunicorn httpbin && \
```
Comment (sivel) [inline Dockerfile]: Looks like `py3-pip` provides a `pip`, but maybe we should use `pip3` to be specific?

**Review by sivel — APPROVED** [review]
A couple of minor comments, this looks good.

File: Dockerfile (hunk 2 above)
Comment (jborean93) [reply Dockerfile]: That's fair, will update

File: Dockerfile (hunk 1 above)
Comment (jborean93) [reply Dockerfile]: Added.

**Pull request merged**